File: Exam/Exam 1/P. Minimize Number.cpp
/*
    ** Sujan Sarkar **
    Github: https://github.com/sujansarkar69
*/
#include <bits/stdc++.h>
using namespace std;
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    int ans = 0, minn = INT_MAX;

    for (int i = 0; i < n; i++)
    {
        ans = 0;
        while (arr[i] % 2 == 0)
        {
            ans++;
            arr[i] /= 2;
        }
        minn = min(ans, minn);
    }

    cout << minn << endl;

    return 0;
}

File: Exam/Exam 1/P. Minimize Number.py
import sys
n = int(input())
a = list(map(int,input().split()))
ans = 0
minn = sys.maxsize
for i in range(0, len(a)):
    ans = 0
    while a[i] % 2 == 0:
        ans += 1
        a[i] /= 2
    minn = min(ans, minn)
print(minn)
